test/e2e/metrics_test.go

```diff
@@ -116,6 +116,113 @@ func TestOperatorControllerMetricsExportedEndpoint(t *testing.T) {
 	require.Contains(t, string(output), "200 OK", "Metrics endpoint did not return 200 OK")
 }
 
+// nolint:gosec
+// TestCatalogdMetricsExportedEndpoint verifies that the metrics endpoint for the catalogd
+// is exported correctly and accessible by authorized users through RBAC and a ServiceAccount token.
+// The test performs the following steps:
+// 1. Creates a ClusterRoleBinding to grant necessary permissions for accessing metrics.
+// 2. Generates a ServiceAccount token for authentication.
+// 3. Deploys a curl pod to interact with the metrics endpoint.
+// 4. Waits for the curl pod to become ready.
+// 5. Executes a curl command from the pod to validate the metrics endpoint.
+// 6. Cleans up all resources created during the test, such as the ClusterRoleBinding and curl pod.
+func TestCatalogdMetricsExportedEndpoint(t *testing.T) {
```

Comment:
With this all put together, is there any common functionality we can extract into helper functions?

Reply:
We definitely can improve that.